src/tls.rs

```diff
@@ -1,6 +1,234 @@
-//! TLS-related helper types
-use core::ffi::{c_char, CStr};
-use core::fmt::Debug;
+//! Type safe abstraction for esp-tls
+
+use std::{
+    ffi::CStr,
+    fmt::{Debug, Display},
+    io::{self, Read, Write},
```

Comment:
Switch to the `io` traits from `embedded-svc` which re-export `embedded-io`. These are no-std compatible. They also have - behind a feature flag - mapping to `std::io::Read` / `std::io::Write`.

Reply:
I switched to `embedded-io` but I couldn't find where it implements the std traits for me. It has a `std` flag but I couldn't find a blanket impl.

Reply:
[Here](https://github.com/embassy-rs/embedded-io/blob/v0.4.0/src/adapters/std_io.rs). Don't be afraid of `embedded-io` being archived. We are for now stuck with version 0.4, but we'll upgrade soon-ish. In the meantime they migrated it to `embedded-hal`, as a sub-project.  The STD adaptors in the `embedded-hal` workspace are in a separate project, but we'll upgrade to these when we migrate to the newer embedded-io (0.5). No rush.